### PR TITLE
LB-802: Last.fm importer user friendliness improvements

### PR DESCRIPTION
**Pull request by jdaok**
# Problem
[LB-802](https://tickets.metabrainz.org/browse/LB-802)

1. https://community.metabrainz.org/t/request-clarify-importer-restart-text/506204
The last.fm importer restart text is vague and should be clarified

2. https://community.metabrainz.org/t/problem-with-import-from-last-fm/454546
When the user's last.fm is private, the error message is unrelated and unhelpful ("The number submitted listens is different from the that Last.fm reports due to an inconsistency in their API, sorry!")

3. https://tickets.metabrainz.org/browse/LB-258
The last.fm username input should remember the username so it doesn't have to be typed repeatedly.

# Solution

1. Updated general restart text
image

2. Added catching for last.fm privacy error and changed error message
image

3. HTML input box should now suggest previous inputted usernames on most browsers
image

- fixed eslint error from my previous commit  (Expected 'this' to be used by class method)

### CODE REVIEW

File: listenbrainz/webserver/static/js/src/LastFMImporter.tsx [hunk 1]
```diff
@@ -151,6 +155,20 @@ export default class LastFmImporter extends React.Component<
     }
   }
 
+  async getUserPrivacy() {
+    // fetch page from Last.fm
+    const { lastfmUsername } = this.state;
+    const url = `${this.lastfmURL}?method=user.getrecenttracks&user=${lastfmUsername}&api_key=${this.lastfmKey}&format=json`;
+    const response = await fetch(encodeURI(url));
```
Comment (MonkeyDo) [inline listenbrainz/webserver/static/js/src/LastFMImporter.tsx:162]: Let's add a try-catch block around this await to catch any errors with that fetch call.

What should happen if the fetch fails ? (for example a temporary network issue)
Once that is decided, you can add a test with a simulated failure of the fetch with something like `window.fetch.mockImplementationOnce(() => Promise.reject("Some error"));`

**Review by MonkeyDo — CHANGES_REQUESTED** [review]
Looks good so far, thanks for opening this PR and continuing to improve the importer :)

**Comment by MonkeyDo** [comment]
Hi @jdaok !
Can I help you with anything on this PR? I'd love to see those improvements merged :)
Let me know if you have any question !

**Comment by jdaok** [comment]
Moved to https://github.com/metabrainz/listenbrainz-server/pull/1417

**Pull request closed without merging**